### PR TITLE
drivers/w5500: fix W5500 link up/down events in interrupt mode

### PR DESCRIPTION
**Pull request by LP-HAW**
### Contribution description

This PR fixes link up/down events in the W5500 driver in interrupt mode. Apparently the W5500 in MACRAW mode does not generate an interrupt when the link goes up or down. Work around this by polling even in interrupt mode.

I noticed that the W5500 interface would not configure an address after a reset until it received a packet. Also, addresses assigned to the interface are not removed and reconfigured when the network cable is unplugged and plugged back in. 

### Testing procedure

1. Build RIOT with W5500 in interrupt mode, e.g.
`USEMODULE="w5500 auto_init_w5500" CFLAGS="-DW5500_PARAM_INT=GPIO_PIN\(0,0\)" make -C examples/gcoap all flash term`

2. Unplug network cable

3. Run `ifconfig`

#### Without this PR:
The interface is down but still assigned an IPv6 address.

#### With this PR:
The interface has no address assigned to it.

### Issues/PRs references

None

### CODE REVIEW

File: drivers/w5500/w5500.c [hunk 1]
```diff
@@ -181,12 +180,11 @@ static int init(netdev_t *netdev)
...
-        dev->timerInstance.callback = extint;
-        dev->timerInstance.arg = dev;
-        ztimer_set(ZTIMER_MSEC, &dev->timerInstance, dev->p.polling_interval_ms);
-    }
-    else {
+    dev->timerInstance.callback = extint;
+    dev->timerInstance.arg = dev;
+    ztimer_set(ZTIMER_MSEC, &dev->timerInstance, dev->p.polling_interval_ms);
+
+    if(gpio_is_valid(dev->p.irq)) {
```
Comment (benpicco) [inline drivers/w5500/w5500.c:187]:
```suggestion
    if (gpio_is_valid(dev->p.irq)) {
```

File: drivers/w5500/w5500.c [hunk 2]
```diff
@@ -104,10 +104,9 @@ static void extint(void *arg)
     w5500_t *dev = (w5500_t *)arg;
 
     netdev_trigger_event_isr(&dev->netdev);
-    if (!gpio_is_valid(dev->p.irq)) {
-        /* restart timer if we are polling */
-        ztimer_set(ZTIMER_MSEC, &dev->timerInstance, dev->p.polling_interval_ms);
-    }
+
+    ztimer_remove(ZTIMER_MSEC, &dev->timerInstance);
```
Comment (benpicco) [inline drivers/w5500/w5500.c:108]: No need to remove the timer before setting it.

File: drivers/w5500/include/w5500_params.h [hunk 3]
```diff
@@ -41,8 +41,8 @@ extern "C" {
 #ifndef W5500_PARAM_INT
 #define W5500_PARAM_INT         GPIO_UNDEF          /**< set to invalid */
 #endif
-#ifndef CONFIG_W5500_POLLING_INTERVAL
-#define CONFIG_W5500_POLLING_INTERVAL 100u          /**< default polling interval 100 ms */
+#ifndef W5500_POLLING_INTERVAL_MS
+#define W5500_POLLING_INTERVAL_MS 100u              /**< default polling interval 100 ms */
```
Comment (benpicco) [inline drivers/w5500/include/w5500_params.h:45]: No need to drop the `CONFIG_` prefix, it's still used to indicate defines that can be overwritten by the user.

Comment (LP-HAW) [reply drivers/w5500/include/w5500_params.h:45]: Why do the other parameters not have the prefix?

Comment (benpicco) [reply drivers/w5500/include/w5500_params.h:45]: One answer is consistency, but a better answer is that they are overwritten by the board, not the user.

**Review by benpicco — APPROVED** [review]

**Pull request merged**